### ISSUE TITLE
Compiled Button component references react as "React", but requires it as "_react"

### ISSUE THREAD

**Issue by brownstein** [socialtables/ST-UI-Toolkit#44]
I suspect this has gone uncaught because React is commonly exposed globally client-side. Pops up when using server-side rendering.

To reproduce:

- Pull this module into a node 4 project via npm install
- Attempt to require within a react template file
- Observe failure
- Navigate to node_modules/@socialtables/st-ui-toolkit/lib/components/Button/Button.js
- Observe referencing bug

**Comment by rovolution** [comment]
thanks @brownstein! We'll look into resolving this

**Comment by rovolution** [comment]
@brownstein could you post a screenshot of the bug being thrown? thanks!

**Comment by brownstein** [comment]
@rovolution Sure! Here's my terminal output (running Authenticator, saml-lib-integration branch).

<img width="948" alt="terminal-react-ref" src="https://cloud.githubusercontent.com/assets/886693/16383037/6981c944-3c51-11e6-9aac-6da1104bf728.png">

More usefully, here's an excerpt from the lib/components/Button/Button found in my node_modules folder:

```
var _react = require("react");
```
```
var Button = (0, _radium2.default)(_class = function (_Component) {
	_inherits(Button, _Component);

	function Button(properties) {
		_classCallCheck(this, Button);

		var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(Button).call(this, properties));

		var style = properties.style;

		var childProps = _objectWithoutProperties(properties, ["style"]); // eslint-disable-line no-unused-vars


		_this._childProps = childProps;
		return _this;
	}

	/**
  * Update the childProps based on the updated properties passed to the button.
  */


	_createClass(Button, [{
		key: "componentWillReceiveProps",
		value: function componentWillReceiveProps(properties) {
			var style = properties.style;

			var childProps = _objectWithoutProperties(properties, ["style"]); // eslint-disable-line no-unused-vars


			this._childProps = childProps;
		}
	}, {
		key: "componentWillUnmount",
		value: function componentWillUnmount() {
			this._childProps = null;
		}
	}, {
		key: "_getStyle",
		value: function _getStyle(styleName, styles) {
			return this.props.style && this.props.style[styleName] || styles[styleName];
		}
	}, {
		key: "render",
		value: function render() {
			var styles = (0, _styles2.default)({ theme: this.props.color, shape: this.props.shape });
			var activeStateStyles = this._getStyle("active", styles);
			var disabledStateStyles = this._getStyle("disabled", styles);

			return React.createElement(
				"button",
				_extends({}, this._childProps, {
					type: this.props.type,
					style: [styles.base, !this.props.disabled && activeStateStyles, this.props.disabled && disabledStateStyles, this.props.style && this.props.style && this.props.style.base],
					onClick: this.props.disabled ? null : this.props.onClick }),
				this.props.children
			);
		}
	}]);

	return Button;
}(_react.Component)) || _class;
```

**Comment by brownstein** [comment]
The current workaround I'm using for server-side rendering is:
```
global.React = React;
```
which although functional is kind of embarrassing.

**Comment by danmactough** [comment]
Ran into this as well. I think the problem is that while some components import `React`, others, just as the `Button` component, do not. The latter will not render server-side without exposing `React` globally.

**Comment by brownstein** [comment]
@danmactough I think everything imports React, but some sub-components reference the global version.

**Comment by danmactough** [comment]
> I think everything imports React, but some sub-components reference the global version.

No. Compare the `Button` component, for example (which doesn't) https://github.com/socialtables/ST-UI-Toolkit/blob/master/src/components/Button/Button.js#L2 with the `TextBox` component (which does) https://github.com/socialtables/ST-UI-Toolkit/blob/master/src/components/TextBox/TextBox.js#L2

**Comment by rovolution** [comment]
Cleaning this up and trying to resolve this. ill let you know once we come up with a fix

**Comment by rovolution** [comment]
Alright I figured out most of the issues and it should work w/out having to bind React to the global namespace.

It was a weird issue during the Babel compilation process where the `transform-react-jsx` plugin automatically transpiles all JSX to `React.createElement()`

Overrode the pragma to `_react.createElement` (since `_react` is the variable that `require('react')` is stored to and that did the trick!

You will still see some Radium issues related to [this](https://github.com/FormidableLabs/radium/issues/759), but the Radium community seems to be on top of it

**Comment by rovolution** [comment]
Published changes to 1.1.0

**Comment by danmactough** [comment]
@rovolution Cool. We'll give it a spin. 🌀

**Comment by rovolution** [comment]
@danmactough radium issue was resolve, so i upgraded and published to v1.1.1

**Comment by danmactough** [comment]
👍
